apps/weather/src/modules/weather/decorators/log-api-response.decorator.ts

```diff
@@ -0,0 +1,29 @@
+import { appendFileSync } from 'fs';
+
+export function LogApiResponse(provider: string) {
+  return function (
+    _target: object,
+    _propertyKey: string,
+    descriptor: PropertyDescriptor
+  ) {
+    const originalMethod = descriptor.value;
+
+    descriptor.value = async function (...args: any[]): Promise<Response> {
+      const logFile = 'logs/weather-providers.log';
+
+      try {
+        const response = await originalMethod.apply(this, args);
+        const clone = response.clone();
+        const body = await clone.text();
+
+        appendFileSync(logFile, `${provider} - Response: ${body}\n`);
```

Comment:
Why are you using the synchronous version here? It blocks the entire application. You should use the async version with void (no await)